### PR TITLE
[metacling] Add missing Transaction for member enumeration:

### LINKED ISSUE #8547: Crash in TDataMember::GetOffsetCint() method
Following simple code crashes:

```cpp
TClass *cl = gROOT->GetClass("ROOT::Experimental::RBox");
TDataMember *m = cl->GetDataMember("fAttrBorder");
printf("offset = %d\n", (int) m->GetOffsetCint());
```

Method is crucial, while used in PyROOT. 
Error trace:
```
#7  0x00007f261b3a1bc7 in cling::DeclCollector::HandleInterestingDecl(clang::DeclGroupRef) () from /home/linev/build/json/lib/libCling.so
#8  0x00007f261bb1b661 in clang::ASTReader::PassInterestingDeclsToConsumer() () from /home/linev/build/json/lib/libCling.so
#9  0x00007f261bb02876 in clang::ASTReader::FindExternalVisibleDeclsByName(clang::DeclContext const*, clang::DeclarationName) () from /home/linev/build/json/lib/libCling.so
#10 0x00007f261bbf3038 in clang::MultiplexExternalSemaSource::FindExternalVisibleDeclsByName(clang::DeclContext const*, clang::DeclarationName) () from /home/linev/build/json/lib/libCling.so
#11 0x00007f261d7d8e5a in clang::DeclContext::lookup(clang::DeclarationName) const () from /home/linev/build/json/lib/libCling.so
#12 0x00007f261baa4c68 in clang::ASTReader::CompleteRedeclChain(clang::Decl const*) () from /home/linev/build/json/lib/libCling.so
#13 0x00007f261bbf2e00 in clang::MultiplexExternalSemaSource::CompleteRedeclChain(clang::Decl const*) () from /home/linev/build/json/lib/libCling.so
#14 0x00007f261d7c5ccf in clang::TagDecl::getMostRecentDeclImpl() () from /home/linev/build/json/lib/libCling.so
#15 0x00007f261d7d8da2 in clang::DeclContext::lookup(clang::DeclarationName) const () from /home/linev/build/json/lib/libCling.so
#16 0x00007f261baa4c68 in clang::ASTReader::CompleteRedeclChain(clang::Decl const*) () from /home/linev/build/json/lib/libCling.so
#17 0x00007f261bbf2e00 in clang::MultiplexExternalSemaSource::CompleteRedeclChain(clang::Decl const*) () from /home/linev/build/json/lib/libCling.so
#18 0x00007f261bafa0cf in clang::ASTReader::FinishedDeserializing() [clone .part.0] () from /home/linev/build/json/lib/libCling.so
#19 0x00007f261bbf330d in clang::MultiplexExternalSemaSource::FinishedDeserializing() () from /home/linev/build/json/lib/libCling.so
#20 0x00007f261d7bf2bc in clang::RecordDecl::field_begin() const () from /home/linev/build/json/lib/libCling.so
#21 0x00007f261d926d30 in clang::ASTContext::getASTRecordLayout(clang::RecordDecl const*) const () from /home/linev/build/json/lib/libCling.so
#22 0x00007f261d926d81 in clang::ASTContext::getASTRecordLayout(clang::RecordDecl const*) const () from /home/linev/build/json/lib/libCling.so
#23 0x00007f261d926770 in clang::ASTContext::getASTRecordLayout(clang::RecordDecl const*) const () from /home/linev/build/json/lib/libCling.so
#24 0x00007f261b2a028a in TClingDataMemberInfo::Offset() () from /home/linev/build/json/lib/libCling.so
#25 0x00007f261f927c4a in TDataMember::GetOffsetCint() const () from /home/linev/build/json/lib/libCore.so
```

### PR DESCRIPTION
**Pull request by Axel-Naumann**
Fixes #8547

### CODE REVIEW

File: core/metacling/test/TClingDataMemberInfoTests.cxx [hunk 1]
```diff
@@ -220,3 +227,14 @@ class Outer {
...
    EXPECT_EQ(dmInnerProtected->Property(), kIsProtected | kIsClass);
 }
+
+#ifdef HAVE_RBOX
+TEST(TClingDataMemberInfo, issue8553)
+{
+   TClass *clBox = nullptr;
+   ASSERT_TRUE(clBox = TClass::GetClass("ROOT::Experimental::RBox"));
+   TDataMember *dmAttrBorder = nullptr;
+   ASSERT_TRUE(dmAttrBorder = clBox->GetDataMember("fAttrBorder"));
```
Comment (linev) [inline core/metacling/test/TClingDataMemberInfoTests.cxx:237]: After https://github.com/root-project/root/pull/8566 member will be renamed `fAttrBorder` -> `border`.

Comment (Axel-Naumann) [reply core/metacling/test/TClingDataMemberInfoTests.cxx:237]: Well I suppose you need this PR for #8566 so you'd have to rename the member in this test in your PR? ;-) Sorry!

Comment (linev) [reply core/metacling/test/TClingDataMemberInfoTests.cxx:237]: No problem, but you have to merge it first

**Review by pcanal — APPROVED** [review]

**Comment by Axel-Naumann** [comment]
Failures known and unrelated.

**Pull request merged**